src/python/zquantum/core/hamiltonian.py

```diff
@@ -141,6 +145,36 @@ def compute_group_variances(
     return np.array(frame_variances)
 
 
+def get_expectation_values_from_rdms_for_qubitoperator_list(
+    interactionrdm: InteractionRDM,
+    qubitoperator_list: List[QubitOperator],
+    sort_terms: bool = False,
+) -> ExpectationValues:
+    """Computes expectation values of Pauli strings in a list of QubitOperator given a 
+       fermionic InteractionRDM from OpenFermion. All the expectation values for the 
+       operators in the list are returned in a single ExpectationValues object in the
+       same order the operators came in.
+
+    Args:
+        interactionrdm: interaction RDM to use for the expectation values
+            computation, as an OF InteractionRDM object
+        qubitoperator_list: List of qubit operators to compute the expectation values for
+            in the form of OpenFermion QubitOperator objects
+        sort_terms: whether or not each input qubit operator needs to be sorted before calculating expectations
+    Returns:
+        expectation values of Pauli strings in all qubit operators as an ExpectationValues object
+    """
+
+    all_expectations = []
+    for qubitoperator in qubitoperator_list:
+        expectations = get_expectation_values_from_rdms(
+            interactionrdm, qubitoperator, sort_terms=sort_terms
+        )
+        all_expectations += list(expectations.values)
+
+    return ExpectationValues(np.asarray(all_expectations))
```

Comment:
That's a good question! I thought about it but it would imply modifying other functions. I think we might have to change the ExpectationValues object later anyway to avoid the issues in https://www.pivotaltracker.com/story/show/176684130

When we take care of the above story we should think of a good way to return these ExpectationValues. My reasoning is also that the list of qubit operators is supposed to be representing the grouping of a single operator, hence the expectation values can live in a single object.

Reply:
Thanks, Jerome! Makes sense!